airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -605,12 +605,12 @@ def ti_heartbeat(
             "Retrieved current task state", state=previous_state, current_hostname=hostname, current_pid=pid
         )
     except NoResultFound:
-        log.error("Task Instance not found")
+        log.error("Task Instance not found in Task Instance table, might have moved to the Task Instance History table")
```

Comment:
```suggestion
        status.HTTP_410_GONE: {"description": "Task Instance no longer exists, it may have moved to the Task Instance History table"},
```